### PR TITLE
ECP: Add Unit Tests for secp224k1

### LINKED ISSUE #7257: Add unit tests for ecp_mod_p224k1()
The task is to add unit tests for ecp_mod_p224k1(). This function reduces the input modulo secp224k1_p. The result of ecp_mod_p224k1() is not always canonical, use mpi_mod_mpi() to get a canonical representative for easier testing.

Only cases where the input is non-negative and at most twice as long as the modulus. (This is because soon these will be replaced by their fixed width _raw counterparts.)

### PR DESCRIPTION
**Pull request by minosgalanakis**
## Description

Resolves #7257 

## Gatekeeper checklist

- [x] **changelog**not required. Extending testing coverage, does not affect library's behaviour
- [x] **backport** not required. This is worked required by a newly developped interface
- [x] **tests** provided

### CODE REVIEW

File: tests/suites/test_suite_ecp.function [hunk 1]
```diff
@@ -1355,6 +1355,89 @@ exit:
...
+    TEST_LE_U(X.n, 448 / biL);
+    TEST_EQUAL(limbs_res, limbs);
+
+    TEST_EQUAL(mbedtls_mpi_mod_modulus_setup(
+                   &m, N, limbs,
+                   MBEDTLS_MPI_MOD_REP_MONTGOMERY), 0);
+
+    TEST_EQUAL(mbedtls_ecp_mod_p224k1(&X), 0);
+    TEST_LE_U(mbedtls_mpi_core_bitlen(X.p, X.n), 224);
+    mbedtls_mpi_mod_raw_fix_quasi_reduction(X.p, &m);
```
Comment (gabor-mezei-arm) [inline tests/suites/test_suite_ecp.function:1429]: As the description of this issue recommend  the `mbedtls_mpi_mod_mpi()` should be used here.

**Review by gabor-mezei-arm — CHANGES_REQUESTED** [review]
The changes for the P192K1 testing is not part of this PR and must be removed.

**Comment by yanesca** [comment]
@minosgalanakis this is a deliverable for this quarter, it should have a `priority-high` label, could you please add it?

**Comment by minosgalanakis** [comment]
I have rebased and addressed the comment. The code is still based on #7424 since it makes resolving conflicts easier and we can test the code as we expect it to come in.

Please only review the [last three commits](https://github.com/Mbed-TLS/mbedtls/pull/7425/files/cec506083b3dbddd52bdf6d22d5b18c4277f5aae..56ca0e3c50ab498eacb9a2c79d1897a5268a8778)

When the pr for `p192k1` is merged, I will rebase it

**Review by paul-elliott-arm — APPROVED** [review]
LGTM

**Pull request merged**